src/test/java/com/example/bookstore/service/BookServiceTest.java

```diff
@@ -0,0 +1,297 @@
+import com.example.bookstore.dto.book.BookDtoWithoutCategoryIds;
+import com.example.bookstore.dto.book.BookSearchParametersDto;
+import com.example.bookstore.dto.book.CreateBookRequestDto;
+import com.example.bookstore.exception.EntityNotFoundException;
+import com.example.bookstore.mapper.BookMapper;
+import com.example.bookstore.model.Book;
+import com.example.bookstore.repository.book.BookRepository;
+import com.example.bookstore.repository.book.BookSpecificationBuilder;
+import com.example.bookstore.service.book.impl.BookServiceImpl;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
+
+@ExtendWith(MockitoExtension.class)
+public class BookServiceTest {
+    private static final Long EXISTING_ID = 1L;
+    private static final Long NON_EXISTING_ID = 100L;
+    @Mock
+    private BookRepository bookRepository;
+    @Mock
+    private BookMapper bookMapper;
+    @Mock
+    private BookSpecificationBuilder bookSpecificationBuilder;
+    @InjectMocks
+    private BookServiceImpl bookService;
+
+    private CreateBookRequestDto createBookRequestDto;
+    private CreateBookRequestDto updateBookRequestDto;
+    private Book book;
+    private Book updatedBook;
+    private BookDto bookDto;
+    private BookDtoWithoutCategoryIds bookDtoWithoutCategoryIds;
+    private BookDto updatedBookDto;
+    private BookSearchParametersDto bookSearchParametersDto;
+
+    @BeforeEach
+    void setUp() {
+        createBookRequestDto = getCreateBookRequestDto();
+        updateBookRequestDto = getUpdateRequestDto();
+        book = getBook();
+        updatedBook = getUpdatedBook();
+        bookDto = getBookDto();
+        bookDtoWithoutCategoryIds = getBookDtoWithoutCategoryIds();
+        updatedBookDto = getUpdatedBookDto();
+        bookSearchParametersDto = getBookSearchParametersDto();
+    }
+
+    @Test
+    @DisplayName("Verify that the save method is working correctly")
+    public void save_ValidCreateBookRequestDto_ReturnsBookDto() {
+        Mockito.when(bookMapper.toModel(createBookRequestDto)).thenReturn(book);
+        Mockito.when(bookRepository.save(book)).thenReturn(book);
+        Mockito.when(bookMapper.toDto(book)).thenReturn(bookDto);
+
+        BookDto expected = bookDto;
+        BookDto actual = bookService.save(createBookRequestDto);
+
+        assertThat(actual).isEqualTo(expected);
+        verify(bookRepository, times(1)).save(book);
```

Comment:
`times(1)` is by default, remove it
check and fix other places
```suggestion
        verify(bookRepository).save(book);
```